src/payment/PaymentForm.jsx

```diff
@@ -1,27 +1,109 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { reduxForm } from 'redux-form';
+import { reduxForm, SubmissionError } from 'redux-form';
 import { FormattedMessage } from '@edx/frontend-i18n';
 
-import CardDetails from './CardDetails';
+import CardDetails, { SUPPORTED_CARD_ICONS } from './CardDetails';
 import CardHolderInformation from './CardHolderInformation';
 
-class PaymentForm extends React.Component {
-  onSubmit(values) {
-    /* istanbul ignore next */
-    alert(JSON.stringify(values)); // eslint-disable-line
+const CardValidator = require('card-validator');
+
+export class PaymentFormComponent extends React.Component {
+  onSubmit = (values) => {
+    const {
+      firstName,
+      lastName,
+      address,
+      city,
+      country,
+      cardNumber,
+      securityCode,
+      cardExpirationMonth,
+      cardExpirationYear,
+    } = values;
+
+    const errors = {
+      ...this.validateRequiredFields({
+        firstName,
+        lastName,
+        address,
+        city,
+        country,
+        cardNumber,
+        securityCode,
+        cardExpirationMonth,
+        cardExpirationYear,
+      }),
+      ...this.validateCardDetails(
+        cardNumber,
+        securityCode,
+        cardExpirationMonth,
+        cardExpirationYear,
+      ),
+    };
+
+    if (Object.keys(errors).length > 0) {
+      throw new SubmissionError(errors);
+    }
+
+    // TODO: implement payment submission
+  };
+
+  validateCardDetails(cardNumber, securityCode, cardExpirationMonth, cardExpirationYear) {
+    const errors = {};
+
+    const { card, isValid } = CardValidator.number(cardNumber);
+    if (cardNumber) {
+      if (!isValid) {
+        errors.cardNumber = 'Invalid card number';
+      } else {
+        if (!Object.keys(SUPPORTED_CARD_ICONS).includes(card.type)) {
+          errors.cardNumber = 'Unsupported card type';
+        }
+        if (securityCode && securityCode.length !== card.code.size) {
+          errors.securityCode = 'Invalid security code';
+        }
+      }
+    }
+
+    const currentMonth = new Date().getMonth() + 1;
+    const currentYear = new Date().getFullYear();
+    if (
+      cardExpirationMonth &&
+      parseInt(cardExpirationMonth, 10) < currentMonth &&
+      parseInt(cardExpirationYear, 10) === currentYear
```

Comment:
What about a card that expired in, say, December 2018?

Reply:
The year dropdown populates with a range from current year to current year + 15. The user will not be able to select a year in the past.